server/test/unit/server/webservices/views/test_permissions.py

```diff
@@ -189,3 +190,21 @@ def test_revoke_from_role(self, mock_factory, mock_resp):
         self.assertTrue(response is mock_resp.return_value)
         mock_factory.role_manager.return_value.remove_permissions_from_role.assert_called_once_with(
             'test', '/v2/some/', [0])
+
+
+class Test__check_invalid_params(unittest.TestCase):
+
+    def test_check_params(self):
+        """
+        Test the missing value is raised if some required params are missing.
+        """
+
+        params = {'login': None, 'resource': None, 'role_id': 'some_role'}
+        try:
+            permissions._check_invalid_params(params)
+        except MissingValue, response:
+            pass
+        else:
+            raise AssertionError("MissingValue should be raised with missing params")
+        self.assertEqual(response.http_status_code, 400)
```

Comment:
I think these assertions should be moved into the `except MissingValue` codepath replacing the `pass` that is currently there now.